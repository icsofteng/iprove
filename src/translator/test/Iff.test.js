const {translate} = require('../../translator/z3')
const {translate: translate_latex} = require('../../translator/latex')

const test_constants = [{ value: 'P', varType: 'Bool' }, { value: 'Q', varType: 'Bool' }]
const test_rules = [
  {
    type: 'binary',
    symbol: 'iff',
    lhs: {
      type: 'literal',
      value: 'P'
    },
    rhs: {
      type: 'literal',
      value: 'Q'
    }
  },
  {
    type: 'literal',
    value: 'P'
  }
]

test('Iff test', () => {
  expect(translate(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})

test('Iff test latex', () => {
  expect(translate_latex(test_rules,test_constants, [], [], [])).toMatchSnapshot()
})