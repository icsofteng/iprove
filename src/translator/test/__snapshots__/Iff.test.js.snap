// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iff test 1`] = `
"(declare-const P Bool)
(declare-const Q Bool)
(assert (iff P Q))
(assert (not P))
(check-sat)"
`;

exports[`Iff test latex 1`] = `
Array [
  "P \\\\Longleftrightarrow Q",
  "P",
]
`;
